### PR TITLE
Meeting minutes for June 2023

### PR DESCRIPTION
**Pull request by tsyesika**
closes: #57

### CODE REVIEW

File: meeting-minutes/2023-06-13.md [hunk 1]
```diff
@@ -0,0 +1,161 @@
...
+  - Mark Miller (mmark) (Agoric)
+  - Dan Connolly (dckc) (Agoric)
+  - Ian Denhardt (zenhack) (Cap'n Proto)
+  - ICRainbow (icrainbow)
+  - Kris Kowal (kris) (Agoric)
+  - Richard Gibson (richard) (Agoric)
+  - Jim 
+  - David Thompson (dave) (Spritely)
+
+# [#57](https://github.com/ocapn/ocapn/issues/57) Agenda
```
Comment (dckc) [inline meeting-minutes/2023-06-13.md:17]:
```suggestion
## [#57](https://github.com/ocapn/ocapn/issues/57) Agenda Review
```

more than 1 top-level heading seems odd to me.

File: meeting-minutes/2023-06-13.md [hunk 2]
```diff
@@ -0,0 +1,161 @@
...
+
+ - [#56](https://github.com/ocapn/ocapn/issues/56) Ordering conversation about serialization/etc
+ - [#6](https://github.com/ocapn/ocapn/issues/6#issuecomment-1589840247) Did DanC find the right API for the interop demo?
+ - [#8](https://github.com/ocapn/ocapn/issues/8) Jessica Tallon demoing test suite status
+ - [#53](https://github.com/ocapn/ocapn/issues/53) 3rd party handoff: should it be general or specific to certain netlayers?
+ 
+# Minutes
```
Comment (dckc) [inline meeting-minutes/2023-06-13.md:24]:
```suggestion
```

the couple lines below were by way of agenda review, I suggest

File: meeting-minutes/2023-06-13.md [hunk 3]
```diff
@@ -0,0 +1,161 @@
...
+
+markm: not as prepared as I'd like on ordering details #56
+
+cwebber: Jessica Tallon could show test suite
+
+## Implementation plans
+
+zenhack: perhaps folks who are about to implement something that might ossify should give a heads-up. nothing from capn-proto
+
+markm: an encodePassable PR is in review. distinguishing -0 from 0 comes up in this [PR](https://github.com/endojs/endo/pull/1626)
```
Comment (dckc) [inline meeting-minutes/2023-06-13.md:34]:
```suggestion
markm: an encodePassable PR is in review. distinguishing -0 from 0 comes up in [endo PR 1626](https://github.com/endojs/endo/pull/1626)
```

https://www.w3.org/WAI/tips/writing/#make-link-text-meaningful

File: meeting-minutes/2023-06-13.md [hunk 4]
```diff
@@ -0,0 +1,161 @@
...
+
+markm: Part of the goal of OCapN is to have a generalized mutually suspicious capability fabric regardless of particular network encoding, so having generalized handoffs useful there
+
+zenhack: I figured we'd have a protocol that might negotiate how to connect, eg a browser might say "I can connect over websockets and webrtc, but I can't connect over blockchain, I'm sorry".  Ideally Agoric and Spritely converge on a shared protocol but capn'proto probably won't, but I'd like to make it where if there's an OCapN handoff between two cap'n Proto vats that they can choose to cut out OCapN, and vice versa.  I think those are my clarifying goals
+
+dave: *describing the tls+tcp netlayer*
+
+kris: very interested in going down this rabbit hole, had a conversation about what non-blockchain protocol would be. We are considering a layering like “TCP + messaging + ed25519 encryption + capability transfer” without TLS, using an elliptic curve for signing and a different elliptic curve for transport opacity.
+
+dckc: Speaking of rabbit holes, point of order
```
Comment (dckc) [inline meeting-minutes/2023-06-13.md:53]:
```suggestion
```

if we're not recording what the point of order was, this note seems out of place. (The point was to see where we were in the agenda and how to proceed without our chair.)

File: meeting-minutes/2023-06-13.md [hunk 5]
```diff
@@ -0,0 +1,161 @@
...
+
+dckc: the OCapN URI refers to the thing actually being tested?
+
+tsyesika: Yes, this URI represents the actual thing being connected to.
+
+tsyesika: In this case it's pulling the greeter object, a `desc:export` for us.  This is just a helper to get the next import object, it increments an integer.  There are no actual objects defined on our side.  We send a message and we know what the reply should look like.  This is just a placeholder, we know what it looks like on our side.
+
+cwebber: It's interesting to look at the test suite because it's like an assembly language for CapTP, it's not like normally how CapTP abstracts over these things for you, it's kind of neat
+
+dckc: Yes Cap'N Proto has similar things in its test suite, it's neat to look at
```
Comment (dckc) [inline meeting-minutes/2023-06-13.md:81]:
```suggestion
Ian: Yes Cap'N Proto has similar things in its test suite, it's neat to look at
```

I didn't make that point; I think @zenhack did.

File: meeting-minutes/2023-06-13.md [hunk 6]
```diff
@@ -0,0 +1,161 @@
...
+
+MarkM: I agree with the way it was just put, we're shooting for that.  That's one reason I keep bringing up EncodePassable.  My hypothesis is that if EncodePassable has the same virtues as something like Syrup but in a binary protocol, but in addition has a rank order preserving protocol, that's a pretty magical property.  If we could agree on encodepassable as a binary data type that would be fantastic
+
+dckc: has that objective passed your radar
+
+cwebber: I think there's two sides, one smaller thing is getting the captp in the room together.  The bigger thing is not just 3 groups interop, but especially building the captp and friends for broad adoption on the internet, the way ecmascript and activitypub have gotten broad adoption.
+
+markm: I agree with that completely, I think that that's why I've been pushing the encoding work
+
+dckc: How does that assist the goal of interoperability?
```
Comment (dckc) [inline meeting-minutes/2023-06-13.md:111]:
```suggestion
dckc: How does agreement on semantics without agreement on concrete syntax assist the goal of interoperability?
```

File: meeting-minutes/2023-06-13.md [hunk 7]
```diff
@@ -0,0 +1,161 @@
...
+
+tsyesika: I think you're right Mark, and I certainly don't diasagree with your assertion we should standardize on abstract syntax, that's important, I think bridges play a role, but when producing a test suite, I would prefer it if it has a concrete syntax.  I think it makes sense for new implementations that haven't been written to speak the language.
+
+zenhack: One thing, we were talking about the notion of 3rd party handoffs and having different lower netlayers pushed down... I'd like to ask people to review.. Cap'N Proto, it's not battle tested to do this, but I'd like to specify a way to do 3rd party handoffs that's agnostic to any netlayer details.  I'd like to ask folks to read that, and that's something floated a few times.  It would probably be useful as a starting point of what it might look like.  It feels like that would be useful
+
+https://github.com/ocapn/ocapn/issues/53#issuecomment-1572439783
+https://github.com/capnproto/capnproto/blob/455824528cd01268d4f9491a0df961c9a3678852/c%2B%2B/src/capnp/rpc.capnp#L1411-L1512
```
Comment (dckc) [inline meeting-minutes/2023-06-13.md:136]: Both Agoric and Spritely agreed to look at rpc.capnp#L1411-L1512 . I'm struggling to come up with a concrete suggestion for how to say that in the notes.

It's partly water under the bridge: @zenhack already turned the "action items" into assignments on #53, which I think is a fine way to keep the ball moving.

**Comment by tsyesika** [comment]
Thanks for your feedback, I've made those changes. I also fixed the month on the initial commit message.

**Pull request merged**